app/src/main/java/com/mybus/MainActivity.java

```diff
@@ -66,9 +79,24 @@ protected void onCreate(Bundle savedInstanceState) {
     @Override
     public void onMapReady(GoogleMap googleMap) {
         mMap = googleMap;
-        // Add a marker in Mar Del Plata and move the camera
-        LatLng latLng = new LatLng(-37.979858, -57.589794);
-        mMap.addMarker(new MarkerOptions().position(latLng).title("Marker in Mar Del Plata"));
-        mMap.animateCamera(CameraUpdateFactory.newLatLngZoom(latLng, 12));
+        locationUpdater.startListening();
+        centerToLastKnownLocation();
     }
+
+    public void centerToLastKnownLocation(){
+        //get the last gps location
+        LatLng lastLocation = locationUpdater.getLastKnownLocation();
+        if (lastLocation != null) {
+            mMap.animateCamera(CameraUpdateFactory.newLatLngZoom(locationUpdater.getLastKnownLocation(), DEFAULT_MAP_ZOOM));
+        }
+        mMap.addMarker(new MarkerOptions().position(lastLocation).title(CURRENT_LOCATION_MARKER));
+    }
+
+    @Override
+    public void onLocationChanged(double lat, double lon) {
+        mMap.clear();
```

Comment:
Clearing the map, will cause removing all markers when the Bus routes are displayed, you should keep the user marker on a variable and move this marker instead.